client/mixins/imageHelper.ts

```diff
@@ -6,37 +6,16 @@
 import Vue from 'vue';
 import { stringify } from 'qs';
 import { BaseItemDto, BaseItemPerson, ImageType } from '@jellyfin/client-axios';
+import { getShapeFromItemType, CardShapes } from '~/utils/items';
 
-type ImageUrlForElementParams = {
-  item?: BaseItemDto;
-  element?: HTMLElement;
-  tag?: string;
-  itemId?: string;
-  maxWidth?: number;
-  maxHeight?: number;
-  quality?: number;
-  limitByWidth?: boolean;
-  backdropIndex?: number;
-  checkParent?: boolean;
-};
+export interface ImageUrlInfo {
```

Comment:
> EDIT: After reviewing this completely, I don't see where are you using anything else that's not url?

That would be because more refactoring is needed, but I wanted to keep it to images only (More is coming for blurhashes in a follow-up)

It's all combined into one function, because we usually need all of them:

* URL to display
* Blurhash for placeholder
* Tag to check if there's an image

This gives us all three in one method, with the same logic for everything. This prevents us from having to copy-paste the image priority logic 3 different times (like it was done before) and has less risk of them getting different if we make change to one and not the other two.

Reply:
This leaves us two options I guess:

* Rename the function to ``getImageInfo`` or something like that. ``getImageUrl`` is unrealistic.
* Split the logic of getting tags, aspect ratios, etc in different functions, like it was done before. This might also simplify the arguments that need to be passed.

I'm fine with either.

Reply:
I'd rather avoid splitting the logic, since it'll lead to a ton of headaches later on.